### PR TITLE
Package registry client

### PR DESCRIPTION
**Pull request by yim-lee**
Continue https://github.com/apple/swift-package-manager/pull/3640 and
polish up the registry client.

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageRegistry/RegistryManager.swift [hunk 1]
```diff
@@ -8,57 +8,67 @@
...
 import struct Foundation.URLComponents
 import struct Foundation.URLQueryItem
 
-import Dispatch
+import TSCBasic
+import TSCUtility
+
+import Basics
+import PackageLoading
+import PackageModel
```
Comment (tomerd) [inline Sources/PackageRegistry/RegistryManager.swift:21]: nit: we started sorting these alphabetically ^^

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageRegistry/RegistryManager.swift [hunk 2]
```diff
@@ -217,60 +302,166 @@ public final class RegistryManager {
...
+            })
+        }
+    }
+
+    private func makeAsync<T>(_ closure: @escaping (Result<T, Error>) -> Void, on queue: DispatchQueue) -> (Result<T, Error>) -> Void {
+        { result in queue.async { closure(result) } }
+    }
+}
+
+public extension RegistryManager {
```
Comment (tomerd) [inline Sources/PackageRegistry/RegistryManager.swift:423]: does this need to be public?

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
ci failures should be solved with https://github.com/apple/swift-package-manager/pull/3865

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Pull request merged**